src/components/router/routerHelpers.js

```diff
@@ -37,30 +118,37 @@ const basePath =
 const getErrorRoute = routes.find(route => route.activateOnError === true) || {};
 
 /**
- * Return an object matching a specific navigation object.
+ * Return an object generated from both generic routes and specific navigation objects.
+ * ID is not passed to "getRouteDetail" to avoid conflicts between routing and
+ * navigation.
  *
  * @param {object} params
  * @param {string} params.id
  * @param {string} params.pathname
  * @param {boolean} params.returnDefault
+ * @param {Array} params.config
  * @returns {object}
  */
-const getNavigationDetail = ({ id = null, pathname = null, returnDefault = false }) => {
-  const defaultItem = returnDefault && navigation.find(item => item.default === true);
-  let navigationItem;
+const getNavRouteDetail = ({ id = null, pathname = null, returnDefault = false, config = routesConfig }) => {
+  let navRouteItem;
 
   if (id) {
-    navigationItem = navigation.find(item => item.id === id);
+    navRouteItem = config.find(item => item.id === id);
```

Comment:
review: in looking at how we call this ID lookup, as part of the platform nav interaction, we can simplfy this into its own functionality, or remove the other functionality contained in `getNavRouteDetail` if we consider dumping these props directly into the router using `generateRoutes`

Reply:
result: we'll maintain this functionality in the short term, and review again once we start implementing product context w/ hooks and the `react-router-dom` hook `useRouteMatch`